File: flexinput/src/main/java/com/lytefast/flexinput/emoji/Emoji.java
package com.lytefast.flexinput.emoji;

import android.support.annotation.DrawableRes;

import java.util.List;


/**
 * @author ${USER_NAME} on 1/10/17.
 */

public class Emoji {

  public final String strValue;
  public final String[] aliases;

  public Emoji(final String strValue, final String[] aliases) {
    this.aliases = aliases;
    this.strValue = strValue;
  }

  public static class EmojiCategory {
    /**
     * String representation of this category.
     */
    public final String name;
    /**
     * Visual representation of this category.
     */
    @DrawableRes public final int icon;
    /**
     * Emojis that are considered part of this category. This does not imply exclusitivity.
     */
    public final List<Emoji> emojis;

    public EmojiCategory(String name, @DrawableRes int icon, List<Emoji> emojis) {
      this.name = name;
      this.icon = icon;
      this.emojis = emojis;
    }
  }
}

File: flexinput/src/main/java/com/lytefast/flexinput/model/Emoji.kt
package com.lytefast.flexinput.model

import android.os.Parcelable
import androidx.annotation.DrawableRes
import kotlinx.android.parcel.Parcelize


/**
 * Simple representation of a unicode emoji.
 *
 * @author Sam Shih
 */
@Parcelize
data class Emoji(
    val strValue: String,
    val aliases: Array<String>
) : Parcelable


@Parcelize
data class EmojiCategory(
    /**
     * String representation of this category.
     */
    val name: String,
    /**
     * Visual representation of this category.
     */
    @DrawableRes
    val icon: Int,
    /**
     * Emojis that are considered part of this category. This does not imply exclusitivity.
     */
    val emojis: List<Emoji>
) : Parcelable
